yt_dlp/extractor/espn.py

```diff
@@ -353,6 +353,14 @@ def _real_extract(self, url):
             if not cookie:
                 self.raise_login_required(method='cookies')
 
+            jwt = re.search(r'=(.*)\|', cookie.value).group(1)
+            import requests
+            id_token = requests.post('https://registerdisney.go.com/jgc/v6/client/ESPN-ONESITE.WEB-PROD/guest/refresh-auth',
+                                     json={
+                                         'refreshToken': json.loads(base64.urlsafe_b64decode(f'{jwt}==='))['refresh_token'],
+                                     },
+                                     ).json()['data']['token']['id_token']
```

Comment:
Is the "jwt" in the cookie actually a JWT? or is it only the payload of a JWT (without a header or signature)?

If it's a full JWT, you could use `jwt_decode_hs256` from utils:

```suggestion
            id_token = self._download_json(
                'https://registerdisney.go.com/jgc/v6/client/ESPN-ONESITE.WEB-PROD/guest/refresh-auth',
                None, headers={'Content-Type': 'application/json'}, data=json.dumps({
                    'refreshToken': jwt_decode_hs256(jwt)['refresh_token'],
                }).encode())['data']['token']['id_token']
```

otherwise I think this should work:

```suggestion
            id_token = self._download_json(
                'https://registerdisney.go.com/jgc/v6/client/ESPN-ONESITE.WEB-PROD/guest/refresh-auth',
                None, headers={'Content-Type': 'application/json'}, data=json.dumps({
                    'refreshToken': json.loads(base64.urlsafe_b64decode(f'{jwt}==='))['refresh_token'],
                }).encode())['data']['token']['id_token']
```

Reply:
JWT is just the payload, so I just copied the part of `jwt_decode_hs256` that doesn't split on `.`

Reply:
Yes, this worked, thanks! Just needed the `Content-Type` header